synapse/handlers/event_auth.py

```diff
@@ -214,15 +214,13 @@ async def check_restricted_join_rules(
             state_ids: The state of the room as it currently is.
             room_version: The room version of the room being joined.
             user_id: The user joining the room.
-            prev_member_event: The current membership event for this user.
+            prev_membership: The current membership state for this user.
 
         Raises:
             AuthError if the user cannot join the room.
         """
         # If the member is invited or currently joined, then nothing to do.
-        if prev_member_event and (
-            prev_member_event.membership in (Membership.JOIN, Membership.INVITE)
-        ):
+        if prev_membership in (Membership.JOIN, Membership.INVITE):
```

Comment:
I can see that we're only using the membership in this function; this looks correct. Any particular reason for making this change though?

If we're making these changes, could we change the docstring to explain what `prev_membership=None` means? I would guess this means that they've never been a member of the room in question?